File: leetcode/shuffleArray/shuffleArray.py
from typing import List

def shuffle(nums: List[int], n: int) -> List[int]:
    for i in range(n):
        y = n + i
        yi = nums[y]
        del nums[y]
        nums.insert(2*i+1, yi)
    return nums


def bettershuffle(self, nums: List[int], n: int) -> List[int]:
        res = []
        for i, j in zip(nums[:n],nums[n:]):
            res += [i,j]
        return res

if __name__ == "__main__":
    abc = [1, 2, 3, 4, 5, 6]
    d = 3
    cba = shuffle(nums=abc, n=d)
    print(cba)

File: leetcode/shuffleArray/shuffleArray.go
package shufflearray

func removeIndex(nums []int, i int) []int {
	return append(nums[:i], nums[i+1:]...)
}

func insertIndex(nums []int, i int, v int) []int {
	nums = append(nums, 0)
	copy(nums[i+1:], nums[i:])
	nums[i] = v
	return nums
}

func shuffle(nums []int, n int) []int {
	for i := 0; i < n; i++ {
		v := nums[n+i]
		nums = removeIndex(nums, n+i)
		nums = insertIndex(nums, 2*i + 1, v)
	}
	return nums
}